spec/springboard/client/uri_spec.rb

```diff
@@ -12,10 +12,35 @@
   end
 
   describe "parse" do
-    it "should return a URI based on the given string" do
-      uri = described_class.parse('/some_path')
-      expect(uri).to be_a(described_class)
-      expect(uri.to_s).to eq('/some_path')
+    describe "when called with a URI object" do
+      it "should return a cloned URI object" do
+        parsed_uri = described_class.parse(uri)
+        expect(uri.class).to eq(parsed_uri.class)
+        expect(uri.object_id).not_to eq(parsed_uri.object_id)
+      end
+    end
+
+    describe "when called with a URI string" do
+      it "should return a URI based on the given string" do
+        uri = described_class.parse('/some_path')
+        expect(uri).to be_a(described_class)
+        expect(uri.to_s).to eq('/some_path')
+      end
+    end
+  end
+
+  describe "dup" do
+    it "should return a duplicate URI object" do
+      dup_uri = uri.dup
+      expect(uri.class).to eq(dup_uri.class)
+      expect(uri.object_id).not_to eq(dup_uri.object_id)
+    end
+
+    it "should include a duplcated Adressable URI object" do
```

Comment:
This is testing the implementation, not the behavior. That makes the test brittle. You should avoid using `instance_variable_get` or any other method that breaks the encapsulation of the object in tests.

Notice that `Addressable` doesn't appear anywhere else in the tests. It's an implementation detail. In fact, this class was introduced as a step towards _removing_ the dependency on Addressable. Ideally when we do that we wouldn't have to change any test implementations.

A better approach here would be to `dup` a `URI`, call a public method that happens to mutate the underlying `@uri` (e.g. `query=`) and then assert that the copy's `to_s` has changed and the original's has not.

I'd probably write this spec as something like this:

```ruby
describe 'when mutating the copy' do
  it 'should not affect the original' do
     # ...
  end
end
```

Also, typo on "duplcated".

Reply:
I agree, that definitely makes sense. I added in that test and removed the one checking instance variables.